code/Makefile

```diff
@@ -46,3 +48,194 @@ tests: ## Run all the tests.
 	$(MAKE) starknet-tests
 	$(MAKE) discovery-tests
 
+##@ Testnet
+
+.PHONY: testnet testnet-mixed testnet-build testnet-down testnet-clean restart-validators test-integration
+
+testnet: testnet-clean ## Start testnet with 4 malachite containers (single network)
+	@echo "Setting up single-network testnet..."
+	@echo "Creating deployment directories..."
+	@mkdir -p ./deployments/volumes/malachite
+
+	@echo "Generating base testnet structure (genesis, keys, etc.)..."
+	@cargo run --bin informalsystems-malachitebft-example-channel -- \
+		testnet \
+		--nodes 4 \
+		--home ./deployments/volumes/malachite
+
+	@echo "Copying single-net configs..."
+	@cp ./single-net-configs/node0-config.toml ./deployments/volumes/malachite/0/config/config.toml
+	@cp ./single-net-configs/node1-config.toml ./deployments/volumes/malachite/1/config/config.toml
+	@cp ./single-net-configs/node2-config.toml ./deployments/volumes/malachite/2/config/config.toml
+	@cp ./single-net-configs/node3-config.toml ./deployments/volumes/malachite/3/config/config.toml
+
+	@echo "Detecting host IP address..."
+	@HOST_IP=$$(hostname -I 2>/dev/null | awk '{print $$1}' || ipconfig getifaddr en0 2>/dev/null || ipconfig getifaddr en1 2>/dev/null || echo "127.0.0.1"); \
```

Comment:
`hostname -I 2>/dev/null | awk '{print $$1}'` returns 0 even thought the first part returns false, as in my laptop. As a result, the IP is empty.

Reply:
fixed in 6b2b7ce89f0e76aa2a779886c7ea9d27668ca6b8. If you want to retry.